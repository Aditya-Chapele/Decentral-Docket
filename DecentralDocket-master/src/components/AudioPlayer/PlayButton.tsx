/*
Copyright 2019-2022 (c) Dappros Ltd, registered in England & Wales, registration number 11455432. All rights reserved.
You may not use this file except in compliance with the License.
You may obtain a copy of the License at https://github.com/dappros/ethora/blob/main/LICENSE.
Note: linked open-source libraries and components may be subject to their own licenses.
*/

import React from "react"
import { StyleSheet, TouchableOpacity } from "react-native"
import FontAwesome from "react-native-vector-icons/FontAwesome"

export default function PlayButton({
  state,
  onPress,
}: {
  state: "play" | "pause"
  onPress: () => void
}) {
  return (
    <TouchableOpacity style={styles.playButtonContainer} onPress={onPress}>
      <FontAwesome name={state} size={25} color="white" />
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  playButtonContainer: {
    // backgroundColor: '#000',
    borderColor: "rgba(93, 63, 106, 0.2)",
    // borderWidth: 16,
    width: 30,
    height: 30,
    borderRadius: 30,
    alignItems: "center",
    justifyContent: "center",
    // marginHorizontal: 32,
    shadowColor: "#5D3F6A",
    // shadowRadius: 30,
    shadowOpacity: 0.5,
  },
})
