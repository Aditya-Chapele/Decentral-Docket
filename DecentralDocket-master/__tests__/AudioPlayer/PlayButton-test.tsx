import React from "react"
import renderer from "react-test-renderer"
import PlayButton from "../../src/components/AudioPlayer/PlayButton"

test("renders correctly", () => {
  const tree = renderer
    .create(
      <PlayButton onPress={() => console.log("play pressed")} state="pause" />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
