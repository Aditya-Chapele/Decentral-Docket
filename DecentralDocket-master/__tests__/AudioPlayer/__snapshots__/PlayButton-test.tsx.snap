// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderColor": "rgba(93, 63, 106, 0.2)",
      "borderRadius": 30,
      "height": 30,
      "justifyContent": "center",
      "opacity": 1,
      "shadowColor": "#5D3F6A",
      "shadowOpacity": 0.5,
      "width": 30,
    }
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "black",
          "fontSize": 25,
        },
        undefined,
        {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>
</View>
`;
